npm.d.ts

```diff
@@ -0,0 +1,439 @@
+	*/
+	man?: string | string[];
+
+	directories?: DirectoryLocations;
+
+	/**
+	Location for the code repository.
+	*/
+	repository?:
+	| string
+	| {
+		type: string;
+		url: string;
+	};
+
+	/**
+	Script commands that are run at various times in the lifecycle of the package. The key is the lifecycle event, and the value is the command to run at that point.
+	*/
+	scripts?: ScriptsDefinition;
+
+	/**
+	Is used to set configuration parameters used in package scripts that persist across upgrades.
+	*/
+	config?: {
+		[configKey: string]: unknown;
+	};
+
+	/**
+	The dependencies of the package.
+	*/
+	dependencies?: Dependency;
+
+	/**
+	Additional tooling dependencies that are not required for the package to work. Usually test, build, or documentation tooling.
+	*/
+	devDependencies?: Dependency;
+
+	/**
+	Dependencies that are skipped if they fail to install.
+	*/
+	optionalDependencies?: Dependency;
+
+	/**
+	Dependencies that will usually be required by the package user directly or via another dependency.
+	*/
+	peerDependencies?: Dependency;
+
+	/**
+	Package names that are bundled when the package is published.
+	*/
+	bundledDependencies?: string[];
+
+	/**
+	Alias of `bundledDependencies`.
+	*/
+	bundleDependencies?: string[];
+
+	resolutions?: Dependency;
+
+	/**
+	Engines that this package runs on.
+	*/
+	engines?: {
+		[engineName: string]: string;
+	};
+
+	/**
+	 * @deprecated
+	 */
+	engineStrict?: boolean;
+
+	/**
+	Operating systems the module runs on.
+	*/
+	os?: LiteralUnion<
+		'darwin' | 'linux' | 'win32' | '!darwin' | '!linux' | '!win32',
+		string
+	>[];
+
+	/**
+	CPU architectures the module runs on.
+	*/
+	cpu?: LiteralUnion<
+		'x64' | 'ia32' | 'arm' | 'mips' | '!x64' | '!ia32' | '!arm' | '!mips',
+		string
+	>[];
+
+	/**
+	If set to `true`, a warning will be shown if package is installed locally. Useful if the package is primarily a command-line application that should be installed globally.
+
+	@deprecated
+	*/
+	preferGlobal?: boolean;
+
+	/**
+	If set to `true`, then npm will refuse to publish it.
+	*/
+	private?: boolean;
+
+	/**
+	 * A set of config values that will be used at publish-time. It's especially handy to set the tag, registry or access, to ensure that a given package is not tagged with 'latest', published to the global public registry or that a scoped module is private by default.
+	 */
+	publishConfig?: {
+		[config: string]: unknown;
+	};
+
+	dist?: {
+		shasum?: string;
+		tarball?: string;
+		integrity?: string;
+		[key: string]: unknown;
+	};
+
+	readme?: string;
+
+	/**
+	An ECMAScript module ID that is the primary entry point to the program.
+	*/
+	module?: string;
+
+	/**
+	A module ID with untranspiled code that is the primary entry point to the program.
+	*/
+	esnext?:
+	| string
+	| {
+		main?: string;
+		browser?: string;
+		[moduleName: string]: string | undefined;
+	};
+
+	/**
+	JSPM configuration.
+	*/
+	jspm?: PackageJson;
+} & {
+	[k: string]: unknown;
```

Comment:
```suggestion
	[key: string]: unknown;
```

Reply:
Why is this in a separate intersection type?

Reply:
See https://github.com/sindresorhus/type-fest/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc#discussion_r265348876